src/lib/components/workspace/Knowledge.svelte

```diff
@@ -0,0 +1,175 @@
+<script lang="ts">
+	import dayjs from 'dayjs';
+	import relativeTime from 'dayjs/plugin/relativeTime';
+	dayjs.extend(relativeTime);
+
+	import { toast } from 'svelte-sonner';
+	import { onMount, getContext } from 'svelte';
+	const i18n = getContext('i18n');
+
+	import { WEBUI_NAME, knowledge } from '$lib/stores';
+
+	import { getKnowledgeItems, deleteKnowledgeById } from '$lib/apis/knowledge';
+
+	import { blobToFile, transformFileName } from '$lib/utils';
+
+	import { goto } from '$app/navigation';
+	import Tooltip from '../common/Tooltip.svelte';
+	import GarbageBin from '../icons/GarbageBin.svelte';
+	import Pencil from '../icons/Pencil.svelte';
+	import DeleteConfirmDialog from '../common/ConfirmDialog.svelte';
+	import ItemMenu from './Knowledge/ItemMenu.svelte';
+
+	let query = '';
+	let selectedItem = null;
+	let showDeleteConfirm = false;
+
+	let filteredItems;
+	$: filteredItems = $knowledge.filter((item) => query === '' || item.name.includes(query));
```

Comment:
Knowledge Search is case sensitive. This would be a fix + i added the description field for search

```
	$: filteredItems = $knowledge.filter(
		(item) =>
			query === '' ||
			item.name.toLowerCase().includes(query.toLowerCase()) ||
			item.description.toLowerCase().includes(query.toLowerCase())
	);

```